repository_service_tuf_worker/repository.py

```diff
@@ -308,29 +315,42 @@ def _bump_and_persist(
         if persist:
             self._persist(role, role_name)
 
-    def _update_timestamp(self, snapshot_version: int) -> Metadata[Timestamp]:
+    def _update_timestamp(
+        self,
+        snapshot_version: Optional[int] = None,
+        skip: Optional[bool] = False,
+    ) -> Metadata[Timestamp]:
         """
         Loads 'timestamp', updates meta info about passed 'snapshot'
         metadata, bumps version and expiration, signs and persists.
 
         Args:
-            snapshot_version: snapshot version to add to new timestamp.
-            db_targets: RSTUFTarget DB objects will be changed as published in
-                the DB SQL.
+            snapshot_version: Optional snapshot version to new timestamp.
+            skip: Skip bumping version and expiration if snpashot
+            version is not provided
         """
         timestamp: Metadata[Timestamp] = self._storage_backend.get(
             Timestamp.type, None
         )
+        if not snapshot_version and skip:
+            logging.debug(
+                "No snapshot version and skip is True. Not bumping Timestamp"
+            )
+            return timestamp
+
         timestamp.signed.snapshot_meta = MetaFile(version=snapshot_version)
 
         self._bump_and_persist(timestamp, Timestamp.type)
 
+        logging.debug("Bumped version of 'Target' role")
         return timestamp
 
     def _update_snapshot(
         self,
         target_roles: Optional[List[str]] = None,
         bump_all: Optional[bool] = False,
+        only_target: Optional[bool] = False,
+        only_snapshot: Optional[bool] = False,
```

Comment:
I am not sure why you need to have both `only_target` and `only_snapshot`?
If `target` is not updated then you don't care even if you load it and do:
```
            snapshot.signed.meta[f"{Targets.type}.json"] = MetaFile(
                version=targets.signed.version
            )
            self._bump_and_persist(snapshot, Snapshot.type)
```
If `target` is changed wouldn't you want to always add the latest version to the new snapshot?

Wouldn't updating target and calling `update_snapshot(only_snapshot=True)` cause an inconsistent state as new snapshot will not contain the new target version in it's meta?

It's better always to load targets in snapshot meta just in case if there is a new version.

Reply:
It is true when update_snapshot was handling only artifacts, but now it handles Targets as a meta for Delegated roles

- When you are adding/removing/updating new Delegations and **all new delegations use offline keys**, you want only to update Target's metadata as the role that handles the delegation's information. (`targets_only`)
py#L881

 - When you are adding/removing/updating new Delegations and **delegations use offline keys and some delegations only online key**, you want only to update Target's metadata as the role that handles the delegation's information. (`targets_only` and `targets_roles`) 
 
 In both cases you cannot update it before calling `_update_timestamp(_update_snapshot...` because you don't know it was the state of the delegated roles
 
Again, a refactoring could help clarify it, but I prefer to do it in a separate and exclusive PR.

Reply:
Okay, as we are limited on time, lets do it in another pr after the conference.